openmrs/src/main/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceImpl.java

```diff
@@ -263,23 +263,29 @@ private Location getLocationByName(String configName, String locationName) {
         List<Observation> observationList = new ArrayList<>();
 
         for (String observationConceptName : observations.keySet()) {
-            Observation observation = new Observation();
+            if (valueIsNotEmpty(observations, observationConceptName)) {
```

Comment:
additionally, please log info if value is empty, that we omit creating this observation because of empty value. I think WARN level will be ok